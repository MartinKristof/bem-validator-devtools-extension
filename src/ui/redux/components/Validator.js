import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Col, Container, Row, Table } from 'reactstrap';
import RulesList from './RulesList';
import Heading from './Heading';
import Icon from './icons/Icon';

const Validator = ({ rules, loading, error, isBemDetected, isValid, inspectElement }) => (
  <Container className="text-center">
    <Row className="my-4">
      <Col>
        {!loading && error && (
          <Fragment>
            <Heading>Any error occured!</Heading>
            <Icon kind="error" />
            <p className="secondary">{error}</p>
          </Fragment>
        )}
        {loading && (
          <Fragment>
            <Heading>Loading...</Heading>
            <Icon kind="loader" />
          </Fragment>
        )}
        {!loading && rules.length > 0 && !error && isBemDetected && !isValid && (
          <div>
            <Heading>Invalid!</Heading>
            <div>
              <Icon kind="invalid" />
              <p className="error">BEM errors found!</p>
            </div>
            <Table responsive striped>
              <thead>
                <tr>
                  <td>Missing class</td>
                  <td>Selector</td>
                </tr>
              </thead>
              <tbody>
                <RulesList rules={rules} inspectElement={inspectElement} />
              </tbody>
            </Table>
          </div>
        )}
        {!loading && !error && isBemDetected && isValid && (
          <Fragment>
            <Heading>Valid!</Heading>
            <div>
              <Icon kind="valid" />
              <p className="success">No BEM errors found!</p>
            </div>
          </Fragment>
        )}
        {!loading && !error && !isBemDetected && !isValid && (
          <Fragment>
            <Heading>No BEM classes detected!</Heading>
            <Icon kind="unknown" />
            <p className="secondary">Nothing to lint!</p>
          </Fragment>
        )}
      </Col>
    </Row>
  </Container>
);

Validator.propTypes = {
  rules: PropTypes.arrayOf(
    PropTypes.exact({ missingClassName: PropTypes.string.isRequired, selector: PropTypes.string.isRequired }),
  ).isRequired,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.string,
  isBemDetected: PropTypes.bool.isRequired,
  isValid: PropTypes.bool.isRequired,
  inspectElement: PropTypes.func.isRequired,
};

export default Validator;
