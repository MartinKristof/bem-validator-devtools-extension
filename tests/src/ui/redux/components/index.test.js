import React from 'react';
import configureMockStore from 'redux-mock-store';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import Validator from '../../../../../src/ui/redux/components';
import stateRecord from '../../../../../src/ui/redux/state';

const mockStore = configureMockStore();

const store = mockStore({ state: stateRecord });

describe('<index />', () => {
  describe('#render', () => {
    it('should render without side-effects', () => {
      const node = mount(
        <Provider store={store}>
          <Validator />
        </Provider>,
      );

      expect(node).toMatchSnapshot();
    });
  });
});
