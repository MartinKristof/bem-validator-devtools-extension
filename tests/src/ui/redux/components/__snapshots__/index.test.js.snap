// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<index /> #render should render without side-effects 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Validator)>
    <Validator
      error={null}
      inspectElement={[Function]}
      isBemDetected={false}
      isValid={false}
      loading={true}
      rules={Array []}
    >
      <Container
        className="text-center"
        tag="div"
      >
        <div
          className="text-center container"
        >
          <Row
            className="my-4"
            tag="div"
          >
            <div
              className="my-4 row"
            >
              <Col
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="col"
                >
                  <Heading>
                    <div>
                      <h4>
                        Loading...
                      </h4>
                    </div>
                  </Heading>
                  <Icon
                    kind="loader"
                    size={64}
                  >
                    <svg
                      height={64}
                      kind="loader"
                      viewBox="0 0 50 50"
                      width={64}
                    >
                      <path
                        d="M25.251 6.461c-10.318 0-18.683 8.365-18.683 18.683h4.068c0-8.071 6.543-14.615 14.615-14.615V6.461z"
                      >
                        <animateTransform
                          attributeName="transform"
                          attributeType="xml"
                          dur="0.6s"
                          from="0 25 25"
                          repeatCount="indefinite"
                          to="360 25 25"
                          type="rotate"
                        />
                      </path>
                    </svg>
                  </Icon>
                </div>
              </Col>
            </div>
          </Row>
        </div>
      </Container>
    </Validator>
  </Connect(Validator)>
</Provider>
`;
